cli/command/system/info.go

```diff
@@ -305,11 +267,73 @@ func printSwarmInfo(dockerCli command.Cli, info types.Info) {
 	}
 }
 
+func printWarnings(dockerCli command.Cli, info types.Info) {
+	if len(info.Warnings) > 0 {
+		fmt.Fprintln(dockerCli.Err(), strings.Join(info.Warnings, "\n"))
+		return
+	}
+	// daemon didn't return warnings. Fallback to old behavior
+	printStorageDriverWarnings(dockerCli, info)
+	printWarningsLegacy(dockerCli, info)
+}
+
+// printWarningsLegacy generates warnings based on information returned by the daemon.
+// DEPRECATED: warnings are now generated by the daemon, and returned in
+// info.Warnings. This function is used to provide backward compatibility with
+// daemons that do not provide these warnings. No new warnings should be added
+// here.
+func printWarningsLegacy(dockerCli command.Cli, info types.Info) {
+	if info.OSType == "windows" {
+		return
+	}
+	if !info.MemoryLimit {
```

Comment:
nit: I know it is a deprecated function, but we could do some little factorization here 😇 
```go
func printWarningLegacy(stdErr io.Writer, property bool, message string){
    if !property{
        fmt.Println(stdErr, "WARNING: "+message)
    }
}

func printWarningsLegacy(dockerCli command.Cli, info types.Info) {
    if info.OSType == "windows" {
        return
    }
    stdErr := dockerCli.Err()
    printWarningLegacy(stdErr, info.MemoryLimit, "No memory limit support")
    printWarningLegacy(stdErr, info. SwapLimit, "No swap limit support")
...
}
```

Reply:
Was considering that at some point (also for the warnings returned by the daemon); at least for the daemon-returned warnings, I decided to print them as-is, that way we could use those warnings for other types of messages (`INFO: your coffee is ready!`, `ERROR: we ran out of milk!`).

We could "engineer" that further (`info.level: info`, `info.message: "your coffee is ready!"`), but that felt like taking it too far


Also noticed there's another warning that possibly could be moved; https://github.com/thaJeztah/cli/blob/02f48b838f1c61e251d7f9eb6128f589acc84949/cli/command/system/info.go#L127-L130, and wasn't sure if there would be others I'd find

Reply:
Right and notice the discreet "tabulation" in this warning message 😄 
And by the way it was only a nit, I'm fine keeping it as is!